cmd/server-main.go

```diff
@@ -612,7 +613,18 @@ func serverMain(ctx *cli.Context) {
 
 	httpServer.TCPOptions.Trace = bootstrapTrace
 	go func() {
-		globalHTTPServerErrorCh <- httpServer.Start(GlobalContext)
+		serveFn, listenErrs, err := httpServer.Init(GlobalContext)
+		if err != nil {
+			globalHTTPServerErrorCh <- err
+			return
+		}
+		for i, e := range listenErrs {
+			if e == nil {
+				continue
+			}
+			logger.LogIf(GlobalContext, fmt.Errorf("Unable to listen on `%s`: %v", listenAddrs[i], listenErrs[i]))
+		}
+		globalHTTPServerErrorCh <- serveFn()
```

Comment:
Pass a `function()` to print the errors if any like this

```
    serve, err := httpServer.Init(GlobalContext, func(listenAddr string, err error) {
    if err != nil {
         logger.LogIf(GlobalContext, fmt.Errorf("Unable to listen on `%s`: %v",  listenAddr, err)
    })
    if err != nil {
        globalHTTPServerErrorCh <- err
	return
    }
    globalHTTPServerErrorCh <- serve()
```

Reply:
Done